torch/testing/_asserts.py

```diff
@@ -157,7 +185,9 @@ def _check_attributes_equal(
     if check_dtype and actual.dtype != expected.dtype:
         return AssertionError(msg_fmtstr.format("dtype", actual.dtype, expected.dtype))
 
-    if check_stride and actual.stride() != expected.stride():
+    if actual.is_sparse != expected.is_sparse:
+        return AssertionError(msg_fmtstr.format("is_sparse", actual.is_sparse, expected.is_sparse))
+    elif not actual.is_sparse and check_stride and actual.stride() != expected.stride():
```

Comment:
Insert:
```
elif actual.is_sparse_csr != expected.is_sparse_csr:
    ...
elif not (actual.is_sparse or actual.is_sparse_csr) and check_stride ...
```

OT to this PR: it may make sense to define strides also for sparse tensors as some sparse tensor algorithms compute the strides anyway when flattened indices are required.